### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by Koniushok**

### CODE REVIEW

File: src/style.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+* {
```
Comment (vik-buchinski) [inline src/style.css]: You shouln't use css files in app at all

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const HtmlWebPackPlugin = require("html-webpack-plugin");
+const path = require("path");
+const merge = require("webpack-merge");
+const HMR = require("./webpack/HMR");
+
+const css = require("./webpack/css.js");
+const extractCSS = require("./webpack/extractCSS.js");
+const devserver = require("./webpack/devserver.js");
+const mailRules = require("./webpack/mainRules.js");
+const image = require("./webpack/image.js");
```
Comment (vik-buchinski) [inline webpack.config.js]: Image? what the image?
Please, use correct naming to be able to understand your code from the first glace.

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  {
+    entry: PATH.sourc + "/index.js",
+    output: { path: PATH.build, filename: "[name].js" },
+    plugins: [htmlPlugin],
+    resolve: {
+      extensions: [".js", ".json", ".jsx"]
+    }
+  },
+  mailRules(),
+  image()
```
Comment (vik-buchinski) [inline webpack.config.js]: Please, add eslint and flowtype plugins as well

File: webpack/css.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module.exports = function() {
```
Comment (vik-buchinski) [inline webpack/css.js]: You won't write css in css files, you'll use styles components. So remove all plugins connected with css

File: package.json [hunk 5]
```diff
@@ -5,10 +5,30 @@
...
   "private": true,
   "dependencies": {
-    "webpack": "^4.28.3",
-    "webpack-cli": "^3.2.1"
+    "react": "^16.7.0",
+    "react-dom": "^16.7.0",
+    "styled-components": "^4.1.3"
   },
   "scripts": {
+    "start": "webpack-dev-server  --mode development  --open ",
```
Comment (vik-buchinski) [inline package.json]: 2 spaces inside script commands? Was it done by guides or by mistake?

File: package.json [hunk 6]
```diff
@@ -5,10 +5,30 @@
...
+    "file-loader": "^3.0.1",
+    "html-webpack-plugin": "^3.2.0",
+    "mini-css-extract-plugin": "^0.5.0",
+    "react-hot-loader": "^4.6.3",
+    "style-loader": "^0.23.1",
+    "webpack": "^4.28.3",
+    "webpack-cli": "^3.2.1",
+    "webpack-dev-server": "^3.1.14",
+    "webpack-merge": "^4.2.1"
   }
```
Comment (vik-buchinski) [inline package.json:72]: Also, please specify versions as on screenshot: http://joxi.ru/brRk40vuJZb80m

File: src/style.css [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+  margin: 0;
+  padding: 0;
+}
+
+a {
+  text-decoration: none;
```
Comment (borkovskij) [inline src/style.css]: use normalize.css instead

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  if (argv.mode === "development") {
+    console.log("development");
+    config = merge([common, css(), devserver(), HMR()]);
+  }
+
+  if (argv.mode === "production") {
+    console.log("production");
+    config = merge([common, extractCSS()]);
+  }
+  //console.log(config);
```
Comment (borkovskij) [inline webpack.config.js]: remove unused comment

File: package.json (hunk 5 above)
Comment (Koniushok) [reply package.json]: ready

File: package.json (hunk 6 above)
Comment (Koniushok) [reply package.json:72]: ready

File: src/style.css (hunk 7 above)
Comment (Koniushok) [reply src/style.css]: ready

File: webpack.config.js (hunk 8 above)
Comment (Koniushok) [reply webpack.config.js]: ready

File: src/style.css (hunk 1 above)
Comment (Koniushok) [reply src/style.css]: ok

File: webpack.config.js (hunk 2 above)
Comment (Koniushok) [reply webpack.config.js]: imageRules

File: package.json [hunk 9]
```diff
@@ -5,10 +5,47 @@
   "author": "Valera <Koniushok@users.noreply.github.com>",
   "private": true,
   "dependencies": {
-    "webpack": "^4.28.3",
-    "webpack-cli": "^3.2.1"
+    "react": "^16.7.0",
+    "react-dom": "^16.7.0",
+    "react-hot-loader": "^4.6.3",
+    "styled-components": "^4.1.3",
+    "webpack": "^4.28.3"
```
Comment (borkovskij) [inline package.json]: Are you sure you need webpack in dependencies instead devDependencies?

File: webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+const htmlPlugin = new HtmlWebPackPlugin({
+  template: "./public/index.html",
+  filename: "index.html",
+  favicon: "./public/favicon.ico"
+});
+const copyPlugin = new CopyWebpackPlugin([
+  {
+    from: path.join(PATH.sourc, "static/normalize.css"),
```
Comment (borkovskij) [inline webpack.config.js]: fix typo error (source)

Comment (Koniushok) [reply webpack.config.js]: thanks, i will change

File: package.json (hunk 9 above)
Comment (Koniushok) [reply package.json]: eslint throw error when webpack in devDependencies. But I can ignore it .
"import/no-extraneous-dependencies": ["error", {"devDependencies": true}]

Comment (vik-buchinski) [reply package.json]: It shouldn't throw it. Do you validate package.json?
You should lint src files only.

Dependencies - it is deps which will be included into build, but you don't need webpack to be included. You need it only to build your project.

File: src/static/normalize.css [hunk 11]
```diff
@@ -0,0 +1,349 @@
+/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
```
Comment (vik-buchinski) [inline src/static/normalize.css]: This file should already be minified.
If you can't find minified file - just minify it using some online css minifier

File: webpack.config.js (hunk 10 above)
Comment (vik-buchinski) [inline webpack.config.js]: Please, copy whole the folder, not only one file

File: package.json (hunk 9 above)
Comment (Koniushok) [reply package.json]: thanks I got it

Comment (Koniushok) [reply package.json]: I didn't use validate package.json.

Comment (Koniushok) [reply package.json]: {
  "valid": true,
  "warnings": [
    "Missing recommended field: keywords",
    "Missing recommended field: contributors"
  ],
  "recommendations": [
    "Missing optional field: homepage"
  ]
}

File: package.json [hunk 12]
```diff
@@ -2,13 +2,73 @@
...
-    "webpack-cli": "^3.2.1"
+    "react": "^16.7.0",
+    "react-dom": "^16.7.0",
+    "react-hot-loader": "^4.6.3",
+    "styled-components": "^4.1.3"
   },
   "scripts": {
-    "build": "webpack --mode production"
+    "start": "webpack-dev-server --mode development --open",
+    "start-hot": "webpack-dev-server --mode development --open --hot",
```
Comment (vik-buchinski) [inline package.json]: Combine please start and star hot commands. You'll always use only 1 comment with hot

File: package.json [hunk 13]
```diff
@@ -2,13 +2,73 @@
...
+    "build": "webpack --mode production",
+    "stylelint": "stylelint src/*.js"
```
Comment (vik-buchinski) [inline package.json]: move stylelint into webpack. It should work in runtime

**Pull request merged**